### ISSUE TITLE
[bug] Dotdrop deleted local copies of config files

### ISSUE THREAD

**Issue by GideonWolfe** [deadc0de6/dotdrop#196]
Dotdrop version (and git commit if run from source): v0.29.5
Using dotdrop: as a submodule, from pypi, '...'

**Describe the bug**

I want to symlink certain configuration files from firefox from my dotdrop directory to my `~/.mozilla` directory. There are only a few files out of the dozens of files and directories firefox creates. 

I import the entire `~/.mozilla` directory and as per the directions on the wiki, start pruning the unnecessary files. I set the link for the `d_mozilla` item to `link_children`. 

I ran `dotdrop install --dry --d_mozilla` and it showed that only three symlinks were being created. When I ran this without the `--dry` flag, my entire `~/.mozilla` directory was gone.  Luckily firefox xan survive that and repopulated with more files. 

the `~/.mozilla/firefox` directory seems to be a symlink for some reason, but none of the files within it that I actually wanted to symlink are symlinked.

Am I missing something? I have successfully done multi level directory symlinks recently without issue  

**Steps to Reproduce**

1. `dotdrop import ~/.mozilla`
2. `cd ~/dotfiles/dotfiles/mozilla`
3. Remove crap I don't need
4. edit `config.yaml` so `link_children` is active
5. `dotdrop install --dry d_mozilla`: output shows three symlinks 
6. `dotdrop install d_mozilla`: output shows "0 dotfiles installed" but actually nuked the folder

**Expected behavior**

I expected only the files remaining in `~/dotfiles/dotfiles/mozilla/` to overwrite the files in `~/.mozilla` with symlinks, leaving the rest untouched. The wiki says 

> " Now all children of the vim dotfile's directory in the dotpath will be symlinked under ~/.vim/ without affecting the rest of the local files, keeping the config file clean and all unwanted files only on the local system. "

**Additional information**

The relevant part of the config file
```yaml
config:
  backup: true
  create: true
  dotpath: dotfiles
dotfiles:
  d_mozilla:
    src: mozilla
    dst: ~/.mozilla
    link: link_children
profiles:
  archlinux:
    dotfiles:
    - d_mozilla

     

```

Dotdrop's execution with the debug logs (`--verbose`)
```bash
$ dotdrop <cmd> --verbose ...
...

[DEBUG][__main__.main] running cmd: install
[DEBUG][dotdrop.templategen.__init__] template additional variables: {'profile': 'archlinux', '_dotdrop_dotpath': '/home/gideon/dotfiles/dotfiles', '_dotdrop_cfgpath': '/home/gideon/dotfiles/config.yaml', '_dotdrop_workdir': '/home/gideon/.config/dotdrop'}
[DEBUG][__main__.cmd_install] execute profile pre actions
[DEBUG][__main__.cmd_install] installing key:"d_mozilla", src:"/home/gideon/dotfiles/dotfiles/mozilla", dst:"/home/gideon/.mozilla", link:"link_children"
[DEBUG][dotdrop.installer.link_children] link_children /home/gideon/dotfiles/dotfiles/mozilla to /home/gideon/.mozilla
[DEBUG][dotdrop.installer.link_children] symlink child /home/gideon/dotfiles/dotfiles/mozilla/firefox to /home/gideon/.mozilla/firefox
[DEBUG][dotdrop.installer._link] ignoring "/home/gideon/.mozilla/firefox", link already exists

```

Any additional information that would help reproduce the bug.

**Comment by deadc0de6** [comment]
Thanks for reporting this. Sorry you're having trouble with dotdrop. 

There are a few points to discuss here. First the debug logs you are showing is the one after you have already run `dotdrop install` once and it is telling you it has already installed the dotfiles (or symlink them in your case). It would be interesting to see the debug logs from the first run (by restoring the original `~/.mozilla` and running `dotdrop install d_mozilla --verbose`). The directory `~/.mozilla/firefox` is not originally a symlink but once you've run dotdrop it is since it symlinks it.

When you use `link_children`, what dotdrop does is looking at the direct children of the dotfile (here `<dotpath>/mozilla`) and taking each of the files/directories it finds and creates a symlink in `~/.mozilla/...` to your `<dotpath>/mozilla/...`). I realise the wiki explanation is not optimal and I will try to re-phrase it to make it more clear.

Also in order to symlink files into your `<dotpath>`, the original files have to be deleted (obviously). When creating symlinks dotdrop **will** explicitely ask you if it can delete the original file to symlink it to the one you have in your `<dotpath>` (something similar to the below):
```
Remove "/home/user/.mozilla/firefox" for link creation? [y/N] ?
```

Using `link_children` in a directory is good if you can isolate the one you want to manage and will be the only one changing them. Here it's not the case as firefox will probably change the content of those directories (`~/.mozilla/firefox`, `~/.mozilla/extensions`, etc). As now even if you symlink `~/.mozilla/firefox` to your dotdrop `<dotpath>` directory, what firefox will do is simply add new files (or change existing ones) in that directory, what will in the end be reflected in your dotdrop `<dotpath>` directory, probably not what you want.

It all comes down to how many files you want to manage with dotdrop but for such a use-case I would treat them a single dotfiles.

On my side I will do the following:
* add a comment in the wiki to always do a backup before attempting such big change (`cp -r ~/.mozilla{,.bak}`)
* update a bit the documentation to make it more clear
* force dotdrop to do a temporary backup when deleting files so that they can be restored in case of issue

If you provide me with more info on what exact files/directories you want to manage in `~/.mozilla`, I can try to help you setup your dotdrop. What are the files you want to manage in `~/.mozilla` ? Can you show me a `ls -R` or `tree` of your `<dotpath>/mozilla`?

**Comment by GideonWolfe** [comment]
Thanks for the quick reply!

I am just a bit confused because I have been using dotdrop for a bit and have set it worked fine.For example, my weechat configuration, I have the following files in `<dotpath>/config/weechat`: `irc.conf`  `plugins.conf` `weechat.conf`

In the config, it is set up like 

```yaml
d_weechat:
    src: config/weechat
    dst: ~/.config/weechat
    link: link_children
```

This allows me to edit the files in `<dotpath/config/weechat` and have the changes reflected into the files in `~/.config/weechat/`, without touching many of the other files.

I originally was using single files, but I was running into an error along the lines of `[ERR] cannot install f_index.html: /home/gideon/.mozilla/firefox is not a file or directory`. 

Basically what I am wanting to do is have three files in my `<dotpath>/mozilla` folder.

1. `userChrome.css` which maps to `~/.mozilla/firefox/<profile folder>/chrome`
2. `index.html` (my custom home page) which maps to `~/.mozilla/firefox` (I then point FF at this file manually)
3. `style.css` which defines the styles for the above file, goes to the same place.

I want to be able to edit these files and have the changes reflected to their actual locations. I have been using the method I described earlier and it has worked fine. Tomorrow I will try again and send logs.

**Comment by deadc0de6** [comment]
It worked great with weechat as these were files and not directories. By using `link_children` on `~/.mozilla` you actually asked dotdrop to symlink all the directories in there.

 I would advise you to simply manage those files individually with dotdrop. You can even use `link: link` on each of these to keep the same kind of behavior you have with weechat.

Also note that you can get the profile path in your config file by using the following:
```yaml
dynvariables:
  mozpath: find ~/.mozilla/firefox -name '*.default'
...
dotfiles:
  f_somefile:
    dst: "{{@@ mozpath @@}}/somefile"
    src: firefox/somefile
...
```

something like this would import the needed files:
```bash
dotdrop import ~/.mozilla/firefox/<profile folder>/chrome/userChrome.css
dotdrop import ~/.mozilla/firefox/index.html
dotdrop import ~/.mozilla/firefox/style.css
```
And then tweak the config to set `link: link` and add the above variable for the firefox profile path.

If anything doesn't work, please dump the config file in here and I'll try to help.

**Comment by GideonWolfe** [comment]
Thanks for the tips. Just to confirm, `link: link` will set it so editing a dotfile in `<dotpath>` will change the files in `.config`, correct?

**Comment by deadc0de6** [comment]
Yes the `link: link` will symlink to your `<dotpath>`. There's one exception: if you're using templating directives it will symlink to `~/.config/dotdrop/` (see <https://github.com/deadc0de6/dotdrop/wiki/symlinked-dotfiles>)
